dimod/discrete/cydiscrete_quadratic_model.pyx

```diff
@@ -51,24 +47,12 @@ cdef class cyDiscreteQuadraticModel:
         if num_cases <= 0:
             raise ValueError("num_cases must be a positive integer")
 
-        cdef VarIndex v = self.adj_.size()  # index of new variable
-
-        self.adj_.resize(v+1)
-
-        cdef Py_ssize_t i
-        for i in range(num_cases):
-            self.bqm_.add_variable()
-
-        self.case_starts_.push_back(self.bqm_.num_variables())
-
-        return v
+        return self.dqm_.add_variable(num_cases) 
```

Comment:
I am returning the value returned by the C function directly, is there any type conversion from c value to python value needed here ?

Reply:
If it's a numeric type then cython will handle it